doc/manpages/dinit-service.5.m4

```diff
@@ -182,13 +182,16 @@ Variable substitution is not performed on the \fBenv\-file\fR property value its
 If the path is not absolute, it is resolved relative to the directory containing the service
 description.
 .TP
-\fBrestart\fR = {yes | true | no | false}
+\fBrestart\fR = {yes | true | on-failure | no | false}
 Indicates whether the service should automatically restart if it stops, including due to
-unexpected process termination or a dependency stopping.
+unexpected process termination or a dependency stopping. "on-failure" is special option
+and indicates auto-restart should happen only when exit code of process is not equal to 0.
 Note that if a service stops due to user request, automatic restart is inhibited.
 $$$changequote(`,')dnl
-ifelse(DEFAULT_AUTO_RESTART, true,
+ifelse(DEFAULT_AUTO_RESTART, ALWAYS,
     ``The default is to automatically restart.'',
+    DEFAULT_AUTO_RESTART, FAIL_ONLY,
+    ``the default is to automatically restart only on non-zero exit code.'',
```

Comment:
What if the process terminates due to a signal?

Reply:
Looks like systemd doesn't restart on SIGHUP, SIGINT, SIGTERM and SIGPIPE signals (Ref: https://www.freedesktop.org/software/systemd/man/latest/systemd.service.html#Restart=).

I think it's reasonable to treat some signals as clean exit status in addition of 0 exit code. I think it could be configurable in build-time with default of SIGHUP, SIGINT, SIGTERM.

Reply:
Treating those signals as special cases is fine if you want to implement that, my point was that the behaviour implemented in the PR right now is to always restart if the process terminates due to a signal, but the documentation here (or in the man page) doesn't say anything about that. The documentation should be clear and complete.

I don't think we should have a build-time option to specify which set of signals count as a clean exit. Too many build-time options makes testing harder and leads to diverging behaviour in deployments which is bad for the user experience.  Just make a reasonable choice and document it (or leave the behaviour as it is, i.e. restart for termination via _any_ signal, but make sure it is documented properly). 

Also, the documentation needs to be clear that this "auto restart" only applies to process-based services, not including `scripted` services (i.e. only to`process` and `bgprocess` services).

Reply:
I will treat `SIGHUP`, `SIGINT`, `SIGUSR1`, `SIGUSR2` and `SIGTERM` as clean exit status because one of use cases of this is on configuration reload. These signals (specially first four ones) are used for daemon config reloads, If that fail daemons usually refuse to run and restarting that daemon doesn't make any benefit.

Reply:
Close this if everything is fixed.